### ISSUE TITLE
Remove custom field templates

### ISSUE THREAD

**Issue by jedateach** [unclecheese/silverstripe-display-logic#32]
Because this module writes its own form field templates, it can break when a theme uses the same template. 

Here's how a formfield extension can update attributes:

``` php
    public function onBeforeRender() {
        if($masters = $this->DisplayLogicMasters()){
            $this->owner->setAttribute("data-display-logic-masters", $masters);
        }
    }
```

**Edit:** Just noticed that this updates the Field, rather than the holder, where it is currently set.

and for the logic itself...I think the criteria could be somehow encoded into JSON, and included via Requirements. This bit requires a bit more effort, although it would mean no more eval, and devs could potentially avoid using jQuery.

**Comment by unclecheese** [comment]
I've made a series of updates that remove the requirement for custom templates. It was a bit tricker than I thought, but I think this will work better.

The one glaring edge case is CheckboxSetField and OptionsetField, as they are not form fields in the traditional sense, and their templates are a bit idiosyncratic. Getting them to work requires that the module use its own template. The tldr; on that is that the JS grabs everything by the "name" attribute, and checkboxes/optionsets have no single field that matches on the name, so the hack is to add the name="MyCheckboxes" attribute to the `<ul>` containing the list of options.

Let me know what you think. This is only in dev-master right now.

**Comment by tractorcow** [comment]
This also breaks the CMS Page Add controller because the custom `CheckboxSetField` overrides necessary styles.

**Comment by tractorcow** [comment]
I think you need to change the JS so that you use data-attributes, if you require a fallback for certain modules. You can extend thesefields in PHP so that you apply these additional attributes.

In particular, the missing `$HolderID` and duplicate `$ID` is what is causing issues.

**Comment by tractorcow** [comment]
I've added a fix in core to address this issue in 3.2;

**Comment by michalkleiner** [comment]
Closing this issue since it's outdated and we're now on Silverstripe CMS 4.12.0 and Display Logic 2.0.5

If the problem persists feel free to open a new bug report and provide replication steps and information on what version of Silverstripe CMS and the module you're using.
